packages/containers/src/AvroViewer/AvroViewer.container.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Map, List } from 'immutable';
+import Component from '@talend/react-components';
+import Toggle from './Toggle.connect';
+export const DEFAULT_STATE = Map({
+	highlighted: List(),
+});
+
+export default class AvroViewer extends React.Component {
+	static displayName = 'Container(AvroViewer)';
+
+	constructor(props) {
+		super(props);
+		this.onSelect = this.onSelect.bind(this);
+	}
+
+	onSelect(event, jsonpath) {
+		const state = this.props.state || DEFAULT_STATE;
+		const adaptedJsonPath = jsonpath
+			.replace(/[-[{}()*+?.,\\^$|#\s]/g, '\\$&')
+			.replace(/\[]/g, '[[0-9]+]');
```

Comment:
Why do you need to convert the JSONPath? Can't we use AvroPath instead?

Reply:
As you said, the format is not really the avro standard but a modified version.
Plus, how we render the nodes is recursive, we don't have the entire record everytime, we don't get the value from root, but from the sub-record we have.